### DISCUSSION TITLE
Sorting groupby result by order of appearance of 1st occurence in group keys

### DISCUSSION THREAD

**Discussion by yohplala** [vaexio/vaex#2209] [Q&A] [answered]
Hi,
To maintain a behavior similar to pandas when `sort=False` in a `groupby`, I would like to sort `groupby` results obtain from vaex using a `sort_key`.

Let's consider this example:
```python
import pandas as pd
import vaex
import numpy as np
from functools import partial

pdf = pd.DataFrame({'group_keys':[1,4,2,4,1], 'val':range(5)})
vdf = vaex.from_pandas(pdf)

def sort_key(val, key_order):
    return np.where(key_order == val)[0][0]

sort_key2 = partial(sort_key, key_order = np.unique(pdf["group_keys"]))

vdf = vaex.from_pandas(pdf)
agg_res = vdf.groupby(by="group_keys", sort=False).agg({"val":"first"})
```

```python
agg_res
Out[34]: 
  #    group_keys    val
  0             1      0
  1             2      2
  2             4      1
```

Please, how to re-order `agg_res` using as sort key `sort_key2`, i.e. that will give me:
```python
pdf.groupby("group_keys", sort=False).agg(**{"val":("val","first")}).reset_index()
Out[36]: 
   group_keys  val
0           1    0
1           4    1
2           2    2
```

Thanks for your help!
Bests,

**Comment by JovanVeljanoski** [comment, marked as answer]
Why don't you do the groupby aggregation in whatever way and sort the results of your `val` column afterwards? Or am I missing something?

**Reply by yohplala** [reply]
Hi @JovanVeljanoski !
Well, in the target use case, `val` will contain unordered values, so it cannot be used for a sort.
**BUT** you mark a point! I could very well add an ordered index in a new column, make a `first` or `last` aggregation on this column, and then sort `groupby` result with this column.
Thanks for your help!
Bests,
